### PR TITLE
Calendar license #184752321

### PR DESCRIPTION
**Pull request by mikewolfd**
### Ticket

[Replace with ticket_id](https://www.pivotaltracker.com/story/show/)

## Code author checklist

- [ ] I have performed a self-review of my code
- [ ] I have created and/or updated relevant documentation, if necessary
- [ ] I have not used any relative imports
- [ ] I have checked for and removed instances of unused config from CMS
- [ ] I have pruned any instances of unused code
- [ ] I have not added any markdown to labels, titles and button text in config

### CODE REVIEW

File: shared/src/license.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export type LicenseStatusItem = {
 
 export type LicenseStatusResult = {
   status: LicenseStatus;
+  expirationISO?: string;
```
Comment (aloverso) [inline shared/src/license.ts:8]: Why should this be potentially undefined? When would it ever not exist but the items and status still do?

File: web/public/mgmt/config.yml [hunk 2]
```diff
@@ -14,6 +14,11 @@ local_backend: true
 media_folder: web/public/img
 logo_url: https://dev.navigator.business.nj.gov/img/Navigator-logo.svg
 public_folder: web/public
+slug:
+  encoding: "ascii"
+  clean_accents: true
+  sanitize_replacement: "_"
```
Comment (aloverso) [inline web/public/mgmt/config.yml:20]: What does this do?

File: web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx [hunk 3]
```diff
@@ -41,10 +43,51 @@ export const FilingsCalendarSingleGrid = (props: Props): ReactElement => {
...
+              <span className="text-no-uppercase text-underline text-base-dark">
+                {licenseEvent.licenseType} {titles[licenseEvent.type]}
+              </span>
+            </a>
+          </Link>
+        </Tag>
+      </div>
+    ) : (
+      <></>
+    );
```
Comment (aloverso) [inline web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx:88]: This seems ripe to be its own component.

Also, we avoid `xx ? <stuff> : <></>` notation in favor of `xx && <stuff>`

File: web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx [hunk 4]
```diff
@@ -41,10 +43,51 @@ export const FilingsCalendarSingleGrid = (props: Props): ReactElement => {
...
 
+  const licenseEvent = getLicenseCalendarEvent(
+    parseDate(props.userData?.licenseData?.expirationISO).add(4, "year").toISOString(),
+    date,
+    props.userData.profileData.industryId
+  );
+
+  if (licenseEvent && visibleFilings.length === NUM_OF_FILINGS_ALWAYS_VIEWABLE) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    remainingFilings.unshift(visibleFilings.pop()!);
```
Comment (aloverso) [inline web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx:58]: What is this doing?

File: web/src/lib/types/types.ts [hunk 5]
```diff
@@ -381,6 +381,24 @@ export interface Filing {
   agency?: TaxAgency | null;
 }
 
+export interface License {
```
Comment (aloverso) [inline web/src/lib/types/types.ts:384]: it seems like we should call this a LicenseEvent or a LicenseFiling... "License" is pretty generic and I don't think it represents what this is.

File: web/src/lib/domain-logic/getLicenseCalendarEvent.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import { parseDate } from "@businessnjgovnavigator/shared/dateHelpers";
+import { LookupIndustryById } from "@businessnjgovnavigator/shared/index";
+import { Dayjs } from "dayjs";
+
+export const getLicenseCalendarEvent = (
+  expirationISO: string | undefined,
+  date: Dayjs,
+  industryId: string | undefined
+): LicenseCalendarEvent | undefined => {
+  if (expirationISO !== undefined && industryId) {
```
Comment (aloverso) [inline web/src/lib/domain-logic/getLicenseCalendarEvent.ts:11]: This should be a guard clause instead

File: web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx [hunk 7]
```diff
@@ -41,10 +43,51 @@ export const FilingsCalendarSingleGrid = (props: Props): ReactElement => {
     );
   });
 
-  const firstTwoFilings = thisMonthFilings.slice(0, NUM_OF_FILINGS_ALWAYS_VIEWABLE);
+  const visibleFilings = thisMonthFilings.slice(0, NUM_OF_FILINGS_ALWAYS_VIEWABLE);
   const remainingFilings = thisMonthFilings.slice(NUM_OF_FILINGS_ALWAYS_VIEWABLE);
   const isOnCurrentYear = getCurrentDate().year().toString() === props.activeYear;
 
+  const licenseEvent = getLicenseCalendarEvent(
+    parseDate(props.userData?.licenseData?.expirationISO).add(4, "year").toISOString(),
```
Comment (aloverso) [inline web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx:51]: Why are we adding 4 years? And why adding it here, in a component, and not in the function? That absolutely seems like domain logic, not render logic.

If the expiration date we're getting from the backend is off by 4 years, shouldn't we even do this on the backend before saving it to begin with?

**Comment by aloverso** [comment]
This is missing all of its frontend tests

File: web/public/mgmt/config.yml (hunk 2 above)
Comment (mikewolfd) [reply web/public/mgmt/config.yml:20]: Stops people from putting weird characters in the slug field, like Unicode stuff, but I can remove that from here

File: web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx (hunk 4 above)
Comment (aloverso) [reply web/src/components/filings-calendar/FilingsCalendarSingleGrid.tsx:58]: Is that part of the story? I didn't see anything about that in the AC

**Pull request merged**